### PR TITLE
Adding order_properties_before to schema

### PR DESCRIPTION
**Pull request by skizhak**
- This is to support ordering Properties of model which extends other models.
- Properties of schema listed under Extends will be ordered after current model properties if they're listed under order_properties_before. so the final order of properties is
properties of extends (not listed in order_properties_before)
current properties
properties of rest of extends model

### CODE REVIEW

**Comment by morrisson** [comment]
At least, you should add brief doc to `schema.md` about this meta property. This feature is ok for me, but I think it's enough and clear for me to write down every properties in propertyOrder list.

**Comment by skizhak** [comment]
yes ill update the doc and add some UT.

**Comment by przemyslaw-dobrowolski-cl** [comment]
@skizhak please do not commit util/bindata.go which is a generated file

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,5 @@ node_modules
 .idea
 etc/gohan.log
 etc/gohan.db
+
+util/bindata.go
```
Comment (przemyslaw-dobrowolski-cl) [inline .gitignore]: bindata.go cannot be ignored. See the comments of the following rejected PR:
https://github.com/cloudwan/gohan/pull/496
and notes
https://github.com/cloudwan/gohan/issues/498

Comment (skizhak) [reply .gitignore]: i got it. but i think this particular commit will require the bindata change to use the new schema field. i will remove the .gitignore but i think i should be committing bindata so people doing 'go get ' install can use this.

**Comment by przemyslaw-dobrowolski-cl** [comment]
@skizhak You're right - in this particular case, bindata must be regenerated and it is correctly added to commit. Approved.

**Comment by skizhak** [comment]
@przemyslaw-dobrowolski-cl thanks. @morrisson @nati gentle reminder on this PR.

**Pull request merged**